lib/model.js

```diff
@@ -3281,14 +3281,14 @@ class Model {
                   value = moment(value).format('YYYY-MM-DD');
                 }
               } else { // go ahread and parse as Date if required
-                if (!(value instanceof Date)) {
+                if (!(value instanceof Date) && value !== Infinity && value !== -Infinity) {
```

Comment:
Too much logic here which should be handled by datatype and is postgres specific. 

//cc @janmeier Should we get dialect date type and call specific method to parse `value` ? This way postgres can do its thing and other dialect can do theirs ?

Reply:
Yep, a parse method on the date type could be a good idea

Reply:
Maybe custom getters and setter for the DATE.prototype?

Reply:
That could work :)